Invalid token error when using a CSharp keyword as variable name

Received following error while compiling

```
Error:
Compiling generated C# code FAILED!
Microsoft (R) Build Engine version 16.7.3+2f374e28e for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
...: error CS1519: Invalid token 'base' in class, struct, or interface member declaration ...
```

Program to repro

```p
machine SimpleMachine {

    var base: int;

    start state Init {
        entry {
            base = 2;
        }
    }
}
```

--- a/Src/PCompiler/CompilerCore/Backend/CSharp/CSharpNameManager.cs
+++ b/Src/PCompiler/CompilerCore/Backend/CSharp/CSharpNameManager.cs
@@ -2,15 +2,26 @@
 using Plang.Compiler.TypeChecker.AST.Declarations;
 using Plang.Compiler.TypeChecker.Types;
 using System.Collections.Generic;
+using System;
 
 namespace Plang.Compiler.Backend.CSharp
 {
     internal class CSharpNameManager : NameManagerBase
     {
         private readonly Dictionary<PLanguageType, string> typeNames = new Dictionary<PLanguageType, string>();
+        private readonly string[] reservedKeywords = new string[]
+        {
+            "bool", "byte", "sbyte", "short", "ushort", "int", "uint", "long", "ulong", "double", "float", "decimal",
+            "string", "char", "void", "object", "typeof", "sizeof", "null", "true", "false", "if", "else", "while", "for", "foreach", "do", "switch",
+            "case", "default", "lock", "try", "throw", "catch", "finally", "goto", "break", "continue", "return", "public", "private", "internal",
+            "protected", "static", "readonly", "sealed", "const", "fixed", "stackalloc", "volatile", "new", "override", "abstract", "virtual",
+            "event", "extern", "ref", "out", "in", "is", "as", "params", "__arglist", "__makeref", "__reftype", "__refvalue", "this", "base",
+            "namespace", "using", "class", "struct", "interface", "enum", "delegate", "checked", "unchecked", "unsafe", "operator", "implicit", "explicit"
+        };
 
         public CSharpNameManager(string namePrefix) : base(namePrefix)
         {
+            Array.Sort(reservedKeywords);
         }
 
         public IEnumerable<PLanguageType> UsedTypes => typeNames.Keys;
@@ -59,6 +70,10 @@ namespace Plang.Compiler.Backend.CSharp
             if (name.StartsWith("$"))
             {
                 name = "TMP_" + name.Substring(1);
+            }
+            else if (Array.BinarySearch(reservedKeywords, name) >= 0)
+            {
+                name = "P_" + name;
             }
 
             return UniquifyName(name);

--- a/Tst/RegressionTests/Feature3Exprs/Correct/issue511/issue511.p
+++ b/Tst/RegressionTests/Feature3Exprs/Correct/issue511/issue511.p
@@ -1,0 +1,10 @@
+machine Main {
+
+    var base: int;
+
+    start state Init {
+        entry {
+            base = 2;
+        }
+    }
+}

